go.mod

```diff
@@ -121,6 +121,7 @@ require (
 	github.com/containerd/fifo v1.1.0 // indirect
 	github.com/containerd/ttrpc v1.2.2 // indirect
 	github.com/containernetworking/cni v1.1.2 // indirect
+	github.com/containers/ocicrypt v1.1.7 // indirect
```

Comment:
Ouch! Lots of new dependencies; do we know if all of those are needed for the functionality that BuildKit needs? At least my eye dropped on this one. I know containerd has https://github.com/containerd/imgcrypt (which also depends on this), but IIRC, support for that was still "experimental" at most.

Perhaps @dmcgowan can fill me in on some of that.

Reply:
Gave a quick look at some of this;

Looks like BuildKit depends on 2 packages from the snapshotter;

- `"github.com/containerd/nydus-snapshotter/pkg/errdefs"`
    - for the `errdefs.IsNotFound()` utility
- `"github.com/containerd/nydus-snapshotter/pkg/converter"`
    - we're using a number of `consts` defined in that package
    - we're using the `Layer`, `MergeOption`, and `UnpackOption` types
    - we're using the `Pack` and `Unpack` functions

It looks like the converter package has options to handle encrypted layers (non-optional; integrated in some functions), so I guess that's where this comes from.

Reply:
cc @jiangliu @taohong PTAL, can we make the package optional in the encrypted nydus layer conversion?

Reply:
Thanks! Yes, mostly looking what parts are absolutely required, and what parts "optional". Maybe some of it _is_ required (and we may have to accept), but as mentioned, ISTR that encrypted images were still somewhat experimental, so looking if it's a "must have" or something we could introduce later.

(Once supported, it's much harder to revert than it is to introduce a new feature later).

Reply:
@thaJeztah Thanks for the detailed instructions! I'm working on [this](https://github.com/containerd/nydus-snapshotter/pull/542).

Reply:
@imeoer Nice; thank you!

If you want to verify the result, feel free to open a (temporary) draft PR in this repository that uses a `replace` rule in the `go.mod` to point at your PR branch (at least, that's something I do at times to verify "does this actually do what I expect it to do", because go modules can be "fun" at time 😂)

Reply:
Awesome, thanks so much @imeoer :tada: @thaJeztah I think https://github.com/containerd/nydus-snapshotter/pull/542 should be good then, which would unblock this PR?

Reply:
Yes! I see the other PR is green with that change as well, so once the Nydus PR is accepted and released / tagged, we can go ahead.

To reduce some of the noise, perhaps we should split some of the related module updates (x/sys, x/crypto, grpc updates) to separate commits (and maybe we should already get those updates on through a separate PR in preparation; WDYT?)

Reply:
Updated: `nydus-snapshotter@v0.13.1` has been released. cc @jedevc @thaJeztah

Reply:
Thank you!

Reply:
Updated :tada: